library/ctr_drbg.c

```diff
@@ -139,9 +159,16 @@ static int block_cipher_df(unsigned char *output,
         return MBEDTLS_ERR_CTR_DRBG_INPUT_TOO_BIG;
     }
 
-    memset(buf, 0, MBEDTLS_CTR_DRBG_MAX_SEED_INPUT +
-           MBEDTLS_CTR_DRBG_BLOCKSIZE + 16);
+    memset(buf, 0, sizeof(buf));
+#if defined(MBEDTLS_CIPHER_C)
+    ret = mbedtls_cipher_setup(&cipher_ctx,
```

Comment:
I think we're supposed to call `init()` before setup. (And also before the first point that might jump to the label where `free()` is called.)